src/devices/Uln2003/Uln2003.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+
+namespace Iot.Device.Uln2003
+{
+    /// <summary>
+    /// This class is for controlling stepper motors that are controlled by a 4 pin controller board.
+    /// </summary>
+    /// <remarks>It is tested and developed using the 28BYJ-48 stepper motor and the ULN2003 driver board.</remarks>
+    public class Uln2003 : IDisposable
+    {
+        /// <summary>
+        /// Default delay in microseconds.
+        /// </summary>
+        private const long StepperMotorDefaultDelay = 1000;
+
+        private int _pin1;
+        private int _pin2;
+        private int _pin3;
+        private int _pin4;
+        private int _steps = 0;
+        private int _engineStep = 0;
+        private int _currentStep = 0;
+        private StepperMode _mode = StepperMode.HalfStep;
+        private bool[,] _currentSwitchingSequence = _halfStepSequence;
+        private bool _isClockwise = true;
+        private GpioController _controller;
+        private Stopwatch _stopwatch;
+        private long _stepMicrosecondsDelay;
+
+        static bool[,] _halfStepSequence = new bool[4, 8] {
+                 { true, true, false, false, false, false, false, true },
+                 { false, true, true, true, false, false, false, false },
+                 { false, false, false, true, true, true, false, false },
+                 { false, false, false, false, false, true, true, true }
+            };
+
+        static bool[,] _fullStepSinglePhaseSequence = new bool[4, 8] {
+                 { true, false, false, false, true, false, false, false },
+                 { false, true, false, false, false, true, false, false },
+                 { false, false, true, false, false, false, true, false },
+                 { false, false, false, true, false, false, false, true }
+            };
+
+        static bool[,] _fullStepDualPhaseSequence = new bool[4, 8] {
+                 { true, false, false, true, true, false, false, true },
+                 { true, true, false, false, true, true, false, false },
+                 { false, true, true, false, false, true, true, false },
+                 { false, false, true, true, false, false, true, true }
+            };
+
+        /// <summary>
+        /// Initialize a Uln2003 class.
+        /// </summary>
+        /// <param name="pin1">The GPIO pin number which corresponds pin A on ULN2003 driver board.</param>
+        /// <param name="pin2">The GPIO pin number which corresponds pin B on ULN2003 driver board.</param>
+        /// <param name="pin3">The GPIO pin number which corresponds pin C on ULN2003 driver board.</param>
+        /// <param name="pin4">The GPIO pin number which corresponds pin D on ULN2003 driver board.</param>
+        public Uln2003(int pin1, int pin2, int pin3, int pin4)
```

Comment:
This constructor should take in a IGpioController object, which is defaulted to null. That way, the binding will have support when used with Gpio extenders.